data-processing-lib/python/src/data_processing/test_support/abstract_test.py

```diff
@@ -96,7 +99,46 @@ def validate_expected_tables(table_list: list[pa.Table], expected_table_list: li
             for j in range(rows1):
                 r1 = t1.take([j])
                 r2 = t2.take([j])
-                assert r1 == r2, f"Row {j} of table {i} are not equal\n\tTransformed: {r1}\n\tExpected   : {r2}"
+                AbstractTest.validate_expected_row(i, j, r1, r2)
+
+    @staticmethod
+    def validate_expected_row(table_index: int, row_index: int, test_row: pa.Table, expected_row: pa.Table):
+        """
+        Compare the two rows for equality, allowing float values to be within a percentage
+        of each other as defined by global _allowed_float_percent_diff.
+        We assume the schema has already been compared and is equivalent.
+        Args:
+            table_index: index of tables that is the source of the rows.
+            row_index:
+            test_row:
+            expected_row:
+        """
+        assert test_row.num_rows == 1, "Invalid usage.  Expected test table with 1 row"
+        assert expected_row.num_rows == 1, "Invalid usage.  Expected expected table with 1 row"
```

Comment:
What exactly are we asserting here? That the table has more than 1 row? Why is this required?

Reply:
only to be sure that the caller (us) is using this method correctly.

Reply:
and the code that follows expects a single row